disnake/state.py

```diff
@@ -784,24 +784,37 @@ def parse_message_create(self, data: gateway.MessageCreateEvent) -> None:
             self._messages.append(message)
         # we ensure that the channel is a type that implements last_message_id
         if channel:
-            if channel.__class__ in (TextChannel, Thread, VoiceChannel):
+            if type(channel) in (TextChannel, Thread, VoiceChannel):
```

Comment:
keep this as `__class__`

Reply:
This change was to avoid two additional `# type: ignore`s, since pyright doesn't support narrowing on `.__class__` :/ (see https://github.com/microsoft/pyright/blob/main/docs/type-concepts.md#type-guards). The difference in terms of performance is negligible (a few nanoseconds), so imo this is worth it, but I'll open a pyright issue in hopes of getting this changed in the future.

Reply:
![image](https://cdn.discordapp.com/emojis/933483831636414464.webp?size=96&quality=lossless)